contracts/rust/src/bin/diff_test.rs

```diff
@@ -443,6 +449,48 @@ fn main() {
             );
             println!("{}", res.encode_hex());
         }
+        Action::GenBLSSig => {
+            let mut rng = jf_utils::test_rng();
+
+            if cli.args.len() != 1 {
+                panic!("Should provide arg1=message");
+            }
+            let message_bytes = cli.args[0].parse::<Bytes>().unwrap();
+
+            // Generate the BLS ver key
+            let key_pair = BLSKeyPair::generate(&mut rng);
+            let vk = key_pair.ver_key();
+            let vk_g2_affine: G2Affine = vk.to_affine();
+            let vk_parsed: ParsedG2Point = vk_g2_affine.into();
+
+            // Sign the message
+            let sig: Signature = key_pair.sign(&message_bytes, CS_ID_BLS_BN254);
+            let sig_affine_point = sig.sigma.into_affine();
+            let sig_parsed: ParsedG1Point = sig_affine_point.into();
+
+            let res = (vk_parsed, sig_parsed);
+            println!("{}", res.encode_hex());
+        }
+        Action::GenBLSHashes => {
+            if cli.args.len() != 1 {
+                panic!("Should provide arg1=message");
+            }
+
+            // Same as in the hash_to_curve function
+            // See https://github.com/EspressoSystems/jellyfish/blob/6c2c08f4e966fd1d454d48bcf30bd41a952f9f76/primitives/src/signatures/bls_over_bn254.rs#L310
+            let hasher_init = &[1u8];
+            let hasher = <DefaultFieldHasher<Keccak256> as HashToField<Fq>>::new(hasher_init);
```

Comment:
This is because the `hasher_init` value is set to [1u8] inside [hash_to_curve](https://github.com/EspressoSystems/jellyfish/blob/9d4073ab8394ca23891f694ad161176dcd688af3/primitives/src/signatures/bls_over_bn254.rs#L315).

Reply:
Ideally in jellyfish, we should have a `hash_to_field()` as well that hides away the initial seed, but this is not a high priority. maybe leave it like this in the code. I'm fine with it.